### ISSUE TITLE
Impossible to migrate to Cypress 10: **/node_modules/** pattern is automatically excluded

### ISSUE THREAD

**Issue by wbramw** [cypress-io/cypress#23611]
### Current behavior

Our current software project is set up in a way where the Cypress test files are in a separate npm package, which gets installed in the actual project where Cypress is launched. This means that the test spec files are located in node_modules folder. This has been working fine, until we wanted to upgrade to Cypress 10.

From the documentation found on https://docs.cypress.io/guides/references/migration-guide#Config-Option-Changes
_The /node_modules/ pattern is automatically added to both e2e.specExcludePattern and component.specExcludePattern, and does not need to be specified (and can't be overridden)._

This seems to make it impossible for us to upgrade to Cypress 10, and leaves us stuck at Cypress version 9.7.0. This looks like a strange decision to remove functionality that was already present in earlier versions of Cypress. Is there any way for us to upgrade anyhow, without completely rearranging all of our repos? Is there a possibility to re-enable this functionality in Cypress?

### Desired behavior

Cypress 10 should allow spec files to be located in node_modules, and should be configurable in the specExcludePattern (or similar) setting.

### Test code to reproduce

https://docs.cypress.io/guides/references/migration-guide#Config-Option-Changes
_The /node_modules/ pattern is automatically added to both e2e.specExcludePattern and component.specExcludePattern, and does not need to be specified (and can't be overridden)._

This option is new in Cypress 10, and is blocking us from upgrading from Cypress 9 to 10.

### Cypress Version

9.7.0 trying to upgrade to 10.x.y

### Node version

v14.16.1, but not Node version-related

### Operating System

Windows 10, but not OS-related

### Other

I already created a discussion topic for this issue (https://github.com/cypress-io/cypress/discussions/23316), but due to lack of response, I decided to submit a bug report.

**Comment by drptbl** [comment]
`supportFile` has the same issue. If you define a path containing `node_modules` in your config, it won't work:
```
Your supportFile is missing or invalid: node_modules/e2e.js
```

To fix it, simply move your config anywhere else and change the path in config. Now it works.

cypress.config.js
```js
  e2e: {
    supportFile: 'node_modules/e2e.js', <<< DOESN'T WORK
    supportFile: 'tests/e2e/e2e.js', <<< WORKS
  },
```

**Comment by wbramw** [comment]
Indeed, supportFile also has this issue. I already moved the supportFile from outside the package into the consuming repo, then Cypress can launch, but it doesn't find any tests. We do not wish to move every single spec file like this, so the issue still exists.

**Comment by thomas-verhoeven** [comment]
We are running into this issue as well. Our support and spec files are maintained in an npm package and installed in multiple projects. The option to disable the automatic exclusion of node_modules would be a much cleaner and safer approach than having to move files around in our build proces or as a post-install script.

**Comment by mackeldeyph** [comment]
Is there any update?
